src/test/kotlin/com/autonomousapps/internal/advice/ConsoleReportTest.kt

```diff
@@ -0,0 +1,305 @@
+  }
+
+  @Test fun `from should return the change advices from a computed analysis`() {
+    // Given
+    val filterSpecBuilder = FilterSpecBuilder()
+    val computedAdvice = ComputedAdvice(
+      unusedDependencies = emptySet(),
+      undeclaredApiDependencies = emptySet(),
+      undeclaredImplDependencies = emptySet(),
+      changeToApi = setOf(orgDotSomethingV1),
+      changeToImpl = setOf(orgDotSomethingV2),
+      filterSpecBuilder= filterSpecBuilder,
+      compileOnlyCandidates = emptySet(),
+      unusedProcs = emptySet()
+    )
+
+    // When
+    var consoleReport = ConsoleReport.from(computedAdvice = computedAdvice)
+
+    // Then
+    val expectedApi = setOf(
+      Advice.change(orgDotSomethingV1, "api")
+    )
+    val actualApi = consoleReport.changeToApiAdvice
+    assertEquals(expectedApi, actualApi, "Expected $expectedApi\nActual   $actualApi\n")
+
+    val expectedImpl = setOf(
+      Advice.change(orgDotSomethingV2, "implementation")
+    )
+    val actualImpl = consoleReport.changeToImplAdvice
+    assertEquals(expectedImpl, actualImpl, "Expected $expectedImpl\nActual   $actualImpl\n")
+  }
+
+  @Test fun `from should not return the change advices from a computed analysis with a filter`() {
+    // Given
+    val filterSpecBuilder = FilterSpecBuilder()
+    filterSpecBuilder.incorrectConfigurationsBehavior = Ignore
+    val computedAdvice = ComputedAdvice(
+      unusedDependencies = emptySet(),
+      undeclaredApiDependencies = emptySet(),
+      undeclaredImplDependencies = emptySet(),
+      changeToApi = setOf(orgDotSomethingV1),
+      changeToImpl = emptySet(),
+      filterSpecBuilder= filterSpecBuilder,
+      compileOnlyCandidates = emptySet(),
+      unusedProcs = emptySet()
+    )
+
+    // When
+    var consoleReport = ConsoleReport.from(computedAdvice = computedAdvice)
+
+    // Then
+    val adviceApi = consoleReport.changeToApiAdvice
+    assertTrue(adviceApi.isEmpty(), "Expected $adviceApi to be empty")
+    val adviceImpl = consoleReport.changeToImplAdvice
+    assertTrue(adviceImpl.isEmpty(), "Expected $adviceImpl to be empty")
+  }
+
+  @Test fun `from should return the compileOnly advices from a computed analysis`() {
+    // Given
+    val filterSpecBuilder = FilterSpecBuilder()
+    val computedAdvice = ComputedAdvice(
+      unusedDependencies = emptySet(),
+      undeclaredApiDependencies = emptySet(),
+      undeclaredImplDependencies = emptySet(),
+      changeToApi = emptySet(),
+      changeToImpl = emptySet(),
+      filterSpecBuilder= filterSpecBuilder,
+      compileOnlyCandidates = setOf(componentV1),
+      unusedProcs = emptySet()
+    )
+
+    // When
+    var consoleReport = ConsoleReport.from(computedAdvice = computedAdvice)
+
+    // Then
+    val expected = setOf(
+      Advice.compileOnly(orgDotSomethingV1, "compileOnly")
+    )
+    val actualApi = consoleReport.compileOnlyDependencies
+    assertEquals(expected, actualApi, "Expected $expected\nActual   $actualApi\n")
```

Comment:
This must fail, because `expected` is a `Set<Advice>`, while `actualApi` is a `Set<Dependency>`. My IDE is warning me about the type mismatch -- doesn't yours?

Reply:
I didn't notice :D.

Reply:
The test was failing for this, but I got a logic problem here that I don't know how to workaround. Can you point to a test of `compileOnly` so that there is a non empty list of advices being computed for this config ?

I don't know what value `componentV1` should have.